bin/src/worker.rs

```diff
@@ -322,30 +323,11 @@ extern "C" {
 }
 
 #[cfg(target_os = "macos")]
-pub unsafe fn get_executable_path() -> String {
-    let capacity = PATH_MAX as usize;
-    let mut temp: Vec<u8> = Vec::with_capacity(capacity);
-    temp.extend(repeat(0).take(capacity));
-    let pathbuf = CString::from_vec_unchecked(temp);
-    let ptr = pathbuf.into_raw();
-
-    let mut size = capacity as u32;
-    if _NSGetExecutablePath(ptr, &mut size) == 0 {
-        let mut temp2: Vec<u8> = Vec::with_capacity(capacity);
-        temp2.extend(repeat(0).take(capacity));
-        let pathbuf2 = CString::from_vec_unchecked(temp2);
-        let ptr2 = pathbuf2.into_raw();
-
-        if libc::realpath(ptr, ptr2) != null_mut() {
-            let path = CString::from_raw(ptr2);
-            path.to_str()
-                .expect("failed to convert CString to String")
-                .to_string()
-        } else {
-            panic!();
-        }
-    } else {
-        panic!("buffer too small");
+pub unsafe fn get_executable_path() -> anyhow::Result<String> {
```

Comment:
we could remove the `unsafe` keyword here

Reply:
I've kept `unsafe` to keep it consistent with other platforms and not change the call site. Do you suggest I remove it just for `macOS` ?

Reply:
yes ! remove if possible!

Reply:
Done.